native/src/core/lib.rs

```diff
@@ -205,11 +204,19 @@ pub mod ffi {
         fn send_fds(socket: i32, fds: &[i32]) -> bool;
         fn recv_fd(socket: i32) -> i32;
         fn recv_fds(socket: i32) -> Vec<i32>;
+        unsafe fn write_to_fd(self: &SuRequest, fd: i32);
 
         #[namespace = "rust"]
         fn daemon_entry();
     }
 
+    // Default constructors
+    extern "Rust" {
+        #[Self = SuRequest]
+        #[cxx_name = "New"]
+        fn default() -> SuRequest;
+    }
```

Comment:
May use https://github.com/dtolnay/cxx/pull/1430/commits/0e8a115ca4b82b39897bfde071336c3b12c23a7a for c++ side constructor